manim/mobject/mobject.py

```diff
@@ -1344,44 +1347,62 @@ def __init__(self, *mobjects, **kwargs):
 
 
 class _AnimationBuilder:
-    def __init__(self, mobject, overridden_animations, generate_target=True):
+    def __init__(self, mobject, generate_target=True, overridden_animation=None):
         self.mobject = mobject
-        self.overridden_animations = overridden_animations
+        self.overridden_animation = overridden_animation
         if generate_target:
             self.mobject.generate_target()
 
     def __getattr__(self, method_name):
         method = getattr(self.mobject.target, method_name)
 
-        if hasattr(method, "_animation_override"):
-            return MethodType(method._animation_override, self.mobject)
-
         def update_target(*method_args, **method_kwargs):
+            if self.overridden_animation:
+                raise NotImplementedError(
+                    "Method chaining is currently not supported for "
+                    "overridden animations"
+                )
+
             if hasattr(method, "_animation_override"):
-                self.overridden_animations.append(
-                    method._animation_override(*method_args, **method_kwargs)
+                self.overridden_animation = method._animation_override(
+                    self.mobject, *method_args, **method_kwargs
                 )
```

Comment:
You shouldn't need to set this as an attribute, right? I'd rather put `overridden_animation = None` before this if block and then return `_AnimationBuilder(..., overridden_animation=overridden_animation)`, or maybe you could set the attribute and then return `self` instead of a new `_AnimationBuilder`, which would also make theoretical subclassing better. Idk.

Reply:
Hmm. By returning `self` instead, we would also eliminate the need for the `generate_target` boolean, because we wouldn't visit `__init__` again anyways. I actually like the idea of that. e8e9ca2e